src/media_observer.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The media integrator relation observer."""
+
+# ignoring duplicate-code with container connect check in the saml observer.
+# pylint: disable=R0801
+
+import logging
+from typing import Optional
+
+import ops
+from charms.data_platform_libs.v0.s3 import CredentialsChangedEvent, S3Requirer
+from ops.framework import Object
+
+import pebble
+import synapse
+from backup_observer import S3_INVALID_CONFIGURATION
+from charm_state import CharmBaseWithState, CharmState
+from charm_types import MediaConfiguration
+from s3_parameters import S3Parameters
+
+logger = logging.getLogger(__name__)
+
+S3_CANNOT_ACCESS_BUCKET = "Media: S3 bucket does not exist or cannot be accessed"
```

Comment:
Same here, potentially move it to a place that the backup can make use of it as well (and not import it via a `media_observer` path).